### PR TITLE
Update to Polars Rust 0.44.2/Python 1.12.0

### LINKED ISSUE #2099: Update to Polars 0.43.1

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #2099

### CODE REVIEW

**Comment by Shoeboxam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opendp/opendp/2144?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2104** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2104?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2144** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2144?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2098** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2098?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2143](https://github.com/opendp/opendp/pull/2143) <a href="https://app.graphite.dev/github/pr/opendp/opendp/2143?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2097** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2097?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2096** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2096?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2095** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2095?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2091** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2091?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2089** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2089?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python/test/test_polars.py [hunk 1]
```diff
@@ -433,6 +433,7 @@ def test_polars_accuracy_threshold():
 
 
 def test_polars_non_wrapping():
+    import io
```
Comment (mccalluc) [inline python/test/test_polars.py:436]: Usually have imports at the top?

File: python/test/test_polars.py [hunk 2]
```diff
@@ -444,11 +445,17 @@ def test_polars_non_wrapping():
...
+
     # for coverage: attribute access works properly
     context.query()._ldf
 
+    # serialize/deserialize roundtrip
+    serial_query = context.query().select(dp.len())
+    serial_query._lf_plan = pl.LazyFrame.deserialize(io.BytesIO(serial_query.serialize())) # type: ignore
+    rel = serial_query.release().collect()
+    assert rel.schema == pl.Schema({"len": pl.UInt32})
+    assert isinstance(rel["len"][0], int)
```
Comment (mccalluc) [inline python/test/test_polars.py:457]: These assertions aren't going the way I'd expect. Is there a reason not to take a query, serialize it, deserialize it, and then compare it to the original... or release both, and make sure they behave the same way?

**Review by mccalluc — APPROVED** [review]
One little question about testing serialization/deserialization, but it doesn't need to block this whole thing. Looks good.

**Pull request merged**